### PR TITLE
Refactor create-release-branch script to a node executable

### PR DESCRIPTION
**Pull request by DaRaFF**
Issues: 
  - https://github.com/upfrontIO/livingdocs-planning/issues/1940
  - https://github.com/upfrontIO/livingdocs-planning/issues/1324

## Description
After using the release-tools for some time, I learned that we need some improvement. Bash scripts are hard to maintain and to test, so I switched to a node cli approach with npx. The advantage of npx is that you don't have to install the package. Instead you can just use npx everywhere and the binaries get downloaded immediately. It's ideal for one time jobs like the release-tools.

### BREAKING CHANGES

* `li-release create-release-branch --base-tag v1.0.1 --release-branch-name release-2017-10 --npm-auth-token <token>` has been replaced with `npx release-tools create-release-branch --base-tag=v1.0.1 --release-branch-name=release-2017-10 --npm-token=<token>`
* `li-release init-release` has been removed
* `li-release finish-release` has been removed

## Create Release Branch (new commands)
* List of available commands: `npx release-tools`
* Create Release Branch (help): `npx release-tools create-release-branch`
* Create Release Branch (simple example): `npx release-tools create-release-branch --base-tag=v1.0.1 --release-branch-name=release-2017-10 --npm-token=<token>`

### CODE REVIEW

File: lib/modules/exec.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module.exports = function (dep) {
+  const { shell, console, colors } = dep
```
Comment (marcbachmann) [inline lib/modules/exec.js]: I don't see any point of injecting modules like that, it makes it a bit unreadable.
I'd just require chalk directly.

**Comment by marcbachmann** [comment]
> The advantage of npx is that you don't have to install the package. Instead you can just use npx everywhere and the binaries get downloaded immediately. It's ideal for one time jobs like the release-tools

Only if you document that properly 😉 . e.g. `npx -p @livingdocs/release-tools create-release-branch --base-tag=v1.0.1 --release-branch-name=release-2017-10 --npm-token=<token>

**Comment by DaRaFF** [comment]
> Only if you document that properly...

When you read the changes properly you will see everything is documented.

**Comment by marcbachmann** [comment]
> When you read the changes properly you will see everything is documented.

I can't any mention of the `--package` option that's required if you run the module "everywhere". Running your scripts is only supported within a git repo, not "everywhere"

**Pull request merged**